src/client/game.cpp

```diff
@@ -2781,10 +2781,12 @@ inline void Game::step(f32 dtime)
 				g_settings->getFloat("fps_max_unfocused") :
 				g_settings->getFloat("fps_max");
 		fps_max = std::max(fps_max, 1.0f);
-		float steplen = 1.0f / fps_max;
+		// Unless you have a barebones game running the server at more than 60 steps
+		// per second is hardly realistic and you're at the point of diminishing returns.
+		fps_max = std::min(fps_max, 60.0f);
```

Comment:
Can we have a setting?

Reply:
I consider this a limitation of the engine and not something users can tune.
If some day someone tunes the timing and server step code so that the server can reasonably run at more than 60fps and that brings real benefits this check should be removed ofc.

Reply:
Why add a hardcoded limitation to 60 Hz though?
Users with >60Hz monitors might want to tune it, for more responsiveness.
A setting would also allow one to decrease it to, say, 10Hz. This is especially useful if you want to test how the game would behave on a dedicated server, just to name one use-case.

Reply:
Can't we base it on the "real" target server step rather than introducing a setting?

Reply:
>Can't we base it on the "real" target server step

What do you mean by that? The code here decides what the target server step is.

Reply:
I was thinking `dedicated_server_step` on a server or `fps_max` on a client, but I see that's what the previous code was already doing (and sfan is just capping `fps_max`, which is in principle a good idea). Disregard my comment then. I don't know whether this warrants a new "client server step" setting.

Reply:
> Why add a hardcoded limitation to 60 Hz though?
> Users with >60Hz monitors might want to tune it, for more responsiveness.

My point is that the server code is (at the moment) not suited to run at more than 60Hz.
For example in my testing the actual dtimes were often much higher.
I'm also not convinced there is any noticable difference between a 60Hz and 144Hz server step, but this is just gut feeling.

> This is especially useful if you want to test how the game would behave on a dedicated server, just to name one use-case.

You can easily run a real server if you for some reason want to test that.

Reply:
>My point is that the server code is (at the moment) not suited to run at more than 60Hz.

Then why cap it at 60 Hz and not lower?
Actual amount of work the server does per step heavily depends on mods. Therefore I'd like to have a setting.

Reply:
Why can't we just use `dedicated_server_step` in this case as well? Heretical?

Reply:
> Then why cap it at 60 Hz and not lower?

Honestly 30 Hz would also be a good value for this (and nobody would notice), but I was expecting more resistance if I  were to change this to <60Hz.

> Therefore I'd like to have a setting.

Adding a setting is equivalent to reverting this change because `fps_max` already exists.

> Why can't we just use `dedicated_server_step` in this case as well? Heretical?

We could, but I think the server can manage more than 11Hz with reasonably optimized games (in singleplayer!) so doing that would get into the "artificial constraint" territory (which I think is reasonable to not want).
OTOH that raises the question on why don't we increase `dedicated_server_step`. Different situation on servers and needs some real-world perf testing IMO.
But that's another topic.

<hr/>

Relevant fact:
Before the step changes the client would feed its dtime directly into the server:
https://github.com/minetest/minetest/blob/cb211d63cded951ea395c03d6afd980accf14bf3/src/client/game.cpp#L1629-L1635
How often the server actually stepped is hard to answer because packet receiving blocks 30ms (= 33 Hz) *unless* a packet was received then it exited earlier:
https://github.com/minetest/minetest/blob/cb211d63cded951ea395c03d6afd980accf14bf3/src/server.cpp#L106-L113

Reply:
Using `dedicated_server_step` in singleplayer would probably make games like mineos unusable with the current default settings.

-------

Kinda related: There's a still unfixed problem with very high fps_max values since Desour's step changes. I usually set a very high fps_max value because MT's FPS limiter is bad and enable VSync instead.

Before the step changes, this worked well. VSync seemed to affect the server step too, probably because of:

> Before the step changes the client would feed its dtime directly into the server

Now, the server step is racing in singleleplayer, causing a lot of entity rubberbanding.

As far as I can see, this PR would fix that (haven't tested), but I'm not sure it's the right approach.

Reply:
I still think this artificial difference between server and singleplayer games is doing more harm than good.
11hz for _server_ side is pretty good, and folks can lower that if wanted/needed (or we could just increase it to 25hz - 0.04).